tutoraspects/templates/aspects/apps/aspects/migrations/alembic/versions/0029_drop_json_column.py

```diff
@@ -0,0 +1,91 @@
+"""
+As part of the Ralph 4.0 upgrade, we drop the unused JSON column "event"
+and rename "event_json" to "event".
+"""
+from alembic import op
+
+
+revision = "0029"
+down_revision = "0028"
+branch_labels = None
+depends_on = None
+on_cluster = " ON CLUSTER '{{CLICKHOUSE_CLUSTER_NAME}}' " if "{{CLICKHOUSE_CLUSTER_NAME}}" else ""
+engine = "ReplicatedReplacingMergeTree" if "{{CLICKHOUSE_CLUSTER_NAME}}" else "ReplacingMergeTree"
+
+# Only used in downgrade, where we have to move data
+upgraded_table_name = "{{ASPECTS_XAPI_DATABASE}}.old_{{ASPECTS_RAW_XAPI_TABLE}}"
+
+def upgrade():
+    op.execute(
+        """
+        SET allow_experimental_object_type=1;
+        """
+    )
+    op.execute(
+        f"""
+        DROP VIEW IF EXISTS {{ ASPECTS_XAPI_DATABASE }}.xapi_events_all_parsed_mv;
```

Comment:
We can't operate on the table as long as this view exists. It doesn't matter on a clean install where the mv doesn't exist yet, but if someone ever upgraded from an older version they would need to re-run dbt to get this back. Right now that's potentially a very expensive operation, but I guess they'd have to run dbt as part of the upgrade anyway. I'm not aware of any reason why people would downgrade / upgrade alembic outside of development.

Reply:
This might be a good time to consider further squashing migrations and resetting the numbering, but I know @Ian2012 is already running some live services of this. 🤔

Reply:
I would need to consider this with my team